src/libs/elektra/keyset.c

```diff
@@ -235,6 +290,7 @@ KeySet * ksVNew (size_t alloc, va_list va)
 	KeySet * keyset = 0;
 
 	keyset = (KeySet *) elektraMalloc (sizeof (KeySet));
+	memset (keyset, 0, sizeof (KeySet));
```

Comment:
Technically, I believe the C standard doesn't mandate `NULL == 0`. So `memset` to zero might not correctly set `NULL` pointers.

Reply:
See 6.3.2.3 Pointers, paragraph 3.

https://www.open-std.org/JTC1/SC22/WG14/www/docs/n1124.pdf

_An integer constant expression with the value 0, or such an expression cast to type `void *`, is called a null pointer constant._

Reply:
You're technically right in that it doesn't mandate the same for `char *`.

Reply:
That's what I meant: https://stackoverflow.com/a/9894047. See also: https://stackoverflow.com/a/61184560.

It is a super technical detail that probably will never have any impact in practice, so I'd just leave it as is.

Reply:
Also, why not use calloc instead of malloc+memset?
https://stackoverflow.com/questions/2688466/why-mallocmemset-is-slower-than-calloc

Reply:
Probably because the `elektraMalloc` code was already there... The new functions already use `elektraCalloc`. Will change this here too.